pystac/extensions/view.py

```diff
@@ -3,44 +3,40 @@
 https://github.com/stac-extensions/view
 """
 
-from typing import Generic, Optional, Set, TypeVar, cast
+from typing import Any, Dict, Generic, Iterable, Optional, Set, TypeVar, cast
 
 import pystac
+from pystac.collection import RangeSummary
 from pystac.extensions.base import (
     ExtensionManagementMixin,
     PropertiesExtension,
+    SummariesExtension,
 )
 from pystac.extensions.hooks import ExtensionHooks
 
 T = TypeVar("T", pystac.Item, pystac.Asset)
 
 SCHEMA_URI = "https://stac-extensions.github.io/view/v1.0.0/schema.json"
+PREFIX = "view:"
 
-OFF_NADIR_PROP = "view:off_nadir"
-INCIDENCE_ANGLE_PROP = "view:incidence_angle"
-AZIMUTH_PROP = "view:azimuth"
-SUN_AZIMUTH_PROP = "view:sun_azimuth"
-SUN_ELEVATION_PROP = "view:sun_elevation"
+OFF_NADIR_PROP = PREFIX + "off_nadir"
+INCIDENCE_ANGLE_PROP = PREFIX + "incidence_angle"
+AZIMUTH_PROP = PREFIX + "azimuth"
+SUN_AZIMUTH_PROP = PREFIX + "sun_azimuth"
+SUN_ELEVATION_PROP = PREFIX + "sun_elevation"
 
 
 class ViewExtension(
     Generic[T], PropertiesExtension, ExtensionManagementMixin[pystac.Item]
 ):
-    """ViewItemExt is the extension of the Item in the View Geometry Extension.
-
-    View Geometry adds metadata related to angles of sensors and other radiance angles
-    that affect the view of resulting data. It will often be combined with other
-    extensions that describe the actual data, such as the eo, sat or sar extensions.
-
-    Args:
-        item : The item to be extended.
-
-    Attributes:
-        item : The Item that is being extended.
-
-    Note:
-        Using ViewItemExt to directly wrap an item will add the 'view' extension ID to
-        the item's stac_extensions.
+    """An abstract class that can be used to extend the properties of an
+    :class:`~pystac.Item` with properties from the :stac-ext:`View Geometry
+    Extension <view>`. This class is generic over the type of STAC Object to be
+    extended (e.g. :class:`~pystac.Item`, :class:`~pystac.Asset`).
+
+    This class will generally not be used directly. Instead, use the concrete
+    implementation associated with the STAC Object you want to extend (e.g.
+    :class:`~ItemViewExtension` to extend an :class:`~pystac.Item`).
```

Comment:
Instead of this text for these types of abstract classes, perhaps we should show an example of using the `ext` method; something like

```suggestion
    To create a concrete instance of ViewExtension, use the :meth:`ViewExtension.ext`
    method, e.g.

    >>> item: pystac.Item = ...
    >>> view_ext = ViewExtension.ext(item)
```

Reply:
Yeah, I like this better. I'll update all of the relevant docstrings.

Reply:
Fixed in a071be035dc3e26cf040c9a534dd0225ee7b5db2